sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -1136,4 +1172,19 @@ private String getLinkName(String sessionId) {
             return existing != null ? existing.getLinkName() : null;
         }
     }
+
+    /**
+     * Map the error to {@link ServiceBusAmqpException}
+     */
+    private Throwable mapError(Throwable throwable, ServiceBusErrorSource errorSource) {
+        if (throwable instanceof ServiceBusAmqpException) {
```

Comment:
This can be condensed.
```java
if ((throwable instanceof ServiceBusAmqpException) || !(throwable instanceof AmqpException)) {
    return throwable;
}

return new ServiceBusAmqpException((AmqpException) throwable, errorSource);
````